pkg/scheduler/schedule_one.go

```diff
@@ -441,13 +441,16 @@ func (sched *Scheduler) schedulePod(ctx context.Context, fwk framework.Framework
 // filter plugins and filter extenders.
 func (sched *Scheduler) findNodesThatFitPod(ctx context.Context, fwk framework.Framework, state *framework.CycleState, pod *v1.Pod) ([]*framework.NodeInfo, framework.Diagnosis, error) {
 	logger := klog.FromContext(ctx)
-	diagnosis := framework.Diagnosis{
-		NodeToStatusMap: make(framework.NodeToStatusMap),
-	}
 
 	allNodes, err := sched.nodeInfoSnapshot.NodeInfos().List()
 	if err != nil {
-		return nil, diagnosis, err
+		return nil, framework.Diagnosis{
+			NodeToStatusMap: make(framework.NodeToStatusMap),
```

Comment:
I don't think `.List` would fail for the snapshot, but couldn't it be left nil?

Reply:
+1 and maybe we can defer the allocation until unsuccess real happens.

Reply:
> couldn't it be left nil?

Ah, yes we can just leave it nil.

@kerthcet Can you elaborate about your proposal?

Reply:
I mean we can make the map (allocate the memory) when pod unschedulable real happens rather than preallocate, especially for 5k nodes, however, scheduling runs in serial, so the benefit is small but will make it the code more complicated, so this is not a good suggestion. 😢